AO3-6240 Use i18n-tasks to rename numbered mailer translation keys

Bilka2 opened the pull request:
# Pull Request Checklist

* [X] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [ ] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [X] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [X] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6240

## Purpose

Rename numbered locale keys used by the mailers.

## Testing Instructions

See comment on Jira issue.

## References

#4497 for moving instructions, will add them to the wiki once this PR is reviewed.

## Credit

Bilka (he/him)

sarken commented on app/views/user_mailer/change_email.html.erb:
```diff
@@ -1,3 +1,3 @@
 <% content_for :message do %>
-  <%= t('.part1', login: style_bold(@user.login), email: style_email(@user.email)).html_safe %>
+  <%= t('.changed', login: style_bold(@user.login), email: style_email(@user.email)).html_safe %>
```
Do we want to do anything about this `html_safe` while we're here? I really can't decide if we should let it go for now or not.

redsummernight replied:
Let's fix that here, we did talk about [using the `html` suffix in keys](https://otwarchive.atlassian.net/browse/AO3-6240?focusedCommentId=359954).

sarken commented on app/views/user_mailer/change_email.text.erb:
```diff
@@ -1,3 +1,3 @@
 <% content_for :message do %>
-<%= t('.part1', login: @user.login, email: @user.email) %>
+<%= t('.changed.html', login: @user.login, email: @user.email) %>
```
Hmm. It _probably_ wouldn't hurt anything for now to use the HTML safe translation in both places, but it'll likely come back to bite us i the future, so could you change this to `.changed.text` and update `en.yml` to include the same text for this key, i.e.,

```
changed:
  html: "%{login}, the email associated with your account has been changed to %{email}"
  text: "%{login}, the email associated with your account has been changed to %{email}"
```

(I don't see any way to easily duplicate the translation for all locales using i18n-tasks, but if you do, that would be even better.)

Bilka2 commented:
Duplicate (copy and rename) locale keys:

* Uncomment the data write config in `config/i18n-tasks.yml` line 25
* Comment the data read config in `config/i18n-tasks.yml` line 18
* Run `bundle exec i18n-tasks data -f yaml | bundle exec i18n-tasks tree-mv -a 'user_mailer.change_email.changed.html' 'user_mailer.change_email.changed.text' -f yaml | bundle exec i18n-tasks data-merge`
* Recomment the data write config in `config/i18n-tasks.yml` line 25
* Uncomment the data read config in `config/i18n-tasks.yml` line 18

The English key needs to be duplicated manually. Don't forget to update any relevant views.

sarken reviewed (approved):
Awesome, thank you so much!
